File: problems/70-Climbing-Stairs/solution.c
int climbStairs(int n){
    if(n == 1) {
        return 1;
    }
    int dp[50];
    memset(dp, 0, 50 * sizeof(int));
    
    dp[n - 1] = 1;
    dp[n - 2] = 2;
    
    for(int i = n - 3; i >= 0; i--) {
        dp[i] = dp[i + 1] + dp[i + 2];
    }
    
    return dp[0];
}

File: problems/70-Climbing-Stairs/solution.go
func climbStairs(n int) int {
	if n == 1 {
		return 1
	}
	steps := make([]int, n)
	steps[n-1] = 1
	steps[n-2] = 2
	for i := n - 3; i >= 0; i-- {
		steps[i] = steps[i+1] + steps[i+2]
	}

	return steps[0]
}

